core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -275,10 +276,34 @@ object ExternalStoreUtils extends Logging {
     }
     val allPoolProps = getAllPoolProperties(url, driver,
       poolProps, hikariCP, isEmbedded)
-    ConnectionProperties(url, driver, dialect, allPoolProps,
+    getConnectionProperties(session, url, driver, dialect, allPoolProps,
       connProps, executorConnProps, hikariCP)
   }
 
+  def getConnectionProperties(session: Option[SparkSession], url: String, driver: String,
+      dialect: JdbcDialect, poolProps: Map[String, String], connProps: Properties,
+      executorConnProps: Properties, hikariCP: Boolean): ConnectionProperties = {
+    if (session.isDefined && session.get.conf.getOption(Attribute.USERNAME_ATTR).isDefined &&
+        session.get.conf.getOption(Attribute.PASSWORD_ATTR).isDefined) {
```

Comment:
@vivekwiz 
Can save the result of conf.getOption and avoid lookup again. Doesn't look like much but still for low latency queries these things can add up. A cleaner scala-like way without compromising perf could be:

```scala
sessionOpt.foreach { session =>
  session.conf.getOption(Attribute.USERNAME_ATTR) match {
    case Some(user) => session.conf.getOption(Attribute.PASSWORD_ATTR) match {
      case Some(password) =>
        ...
      case None =>
    }
    case None =>
  }
}
```